api/src/utils/aggregate-utils.ts

```diff
@@ -0,0 +1,79 @@
+import { IPitch, IUser, IUserAggregate, IPitchAggregate } from 'ssw-common';
+
+import Pitch, { PitchSchema } from '../models/pitch';
+import User, { UserSchema } from '../models/user';
+
+import { santitizePitch, santitizeUser } from './helpers';
+
+const simplifyUser = (user: IUser): Partial<IUser> => ({
+  firstName: user.firstName,
+  preferredName: user.preferredName,
+  lastName: user.lastName,
+  profilePic: user.profilePic,
+  _id: user._id,
+});
+
+const aggregatePitch = async (
+  rawPitch: PitchSchema,
+): Promise<IPitchAggregate> => {
+  const author = simplifyUser(await User.findById(rawPitch.author));
+  const reviewer = simplifyUser(await User.findById(rawPitch.approvedBy));
+
+  const assignmentContributors = await Promise.all(
+    rawPitch.assignmentContributors.map(async (contributor) => ({
+      user: simplifyUser(await User.findById(contributor.userId)),
+      teams: contributor.teams,
+    })),
+  );
+
+  const pendingContributors = await Promise.all(
+    rawPitch.pendingContributors.map(async (contributor) => ({
+      user: simplifyUser(await User.findById(contributor.userId)),
+      teams: contributor.teams,
+    })),
+  );
+
+  const aggregatedPitch = {
+    ...santitizePitch(rawPitch),
+    aggregated: {
+      author: author,
+      reviewedBy: reviewer,
+      assignmentContributors: assignmentContributors,
+      pendingContributors: pendingContributors,
+    },
+  };
+
+  return aggregatedPitch;
+};
+
+const simplifyPitch = (pitch: IPitch): Partial<IPitch> => ({
+  title: pitch.title,
+  description: pitch.description,
+  author: pitch.author,
+});
+
+const aggregateUser = async (rawUser: UserSchema): Promise<IUserAggregate> => {
+  const claimedPitches = await Promise.all(
+    rawUser.claimedPitches.map(async (id) =>
+      simplifyPitch(await Pitch.findById(id)),
+    ),
+  );
+
+  const submittedPitches = await Promise.all(
+    rawUser.submittedPitches.map(async (id) =>
+      simplifyPitch(await Pitch.findById(id)),
+    ),
+  );
```

Comment:
Is there any reason these two promises must be awaited one after another? Since they're independent, I'm wondering if they can be promise.all()'d too.

Reply:
Are you suggesting putting these into a single Promise.all() ? Not sure what that would look like, could you provide an example of what you're talking about :)